spec/helpers/pageflow/linkmap_page/areas_helper_spec.rb

```diff
@@ -47,148 +47,148 @@ module LinkmapPage
+          html = helper.linkmap_areas_div(@entry, configuration)
 
           expect(html).to have_selector('a[data-color-map-component-id="1:aaa"]')
         end
 
         it 'uses color map component id if present, preceding mask perma id' do
-          entry = create(:entry)
-          color_map_file_1 = create(:color_map_file)
-          color_map_file_2 = create(:color_map_file)
+          @entry = PublishedEntry.new(create(:entry, :published))
+          color_map_file_1 = create(:used_file, model: :color_map_file, revision: @entry.revision)
+          color_map_file_2 = create(:used_file, model: :color_map_file, revision: @entry.revision)
           configuration = {
-            'linkmap_areas' => [{'color_map_component_id' => "#{color_map_file_2.id}:aaa",
-                                 'mask_perma_id' => "#{color_map_file_1.id}:aaa"}]
+            'linkmap_areas' => [{'color_map_component_id' => "#{color_map_file_2.perma_id}:aaa",
+                                 'mask_perma_id' => "#{color_map_file_1.perma_id}:aaa"}]
           }
 
-          html = helper.linkmap_areas_div(entry, configuration)
+          html = helper.linkmap_areas_div(@entry, configuration)
 
-          expect(html).to have_selector("a[data-color-map-component-id='#{color_map_file_2.id}:aaa']")
+          expect(html).to have_selector("a[data-color-map-component-id='#{color_map_file_2.perma_id}:aaa']")
         end
 
         it 'uses mask perma id as fallback if present and color map component id is blank' do
-          entry = create(:entry)
-          color_map_file = create(:color_map_file)
+          @entry = PublishedEntry.new(create(:entry, :published))
+          color_map_file = create(:used_file, model: :color_map_file, revision: @entry.revision)
           configuration = {
-            'linkmap_areas' => [{'mask_perma_id' => "#{color_map_file.id}:aaa"}]
+            'linkmap_areas' => [{'mask_perma_id' => "#{color_map_file.perma_id}:aaa"}]
           }
 
-          html = helper.linkmap_areas_div(entry, configuration)
+          html = helper.linkmap_areas_div(@entry, configuration)
 
-          expect(html).to have_selector("a[data-color-map-component-id='#{color_map_file.id}:aaa']")
+          expect(html).to have_selector("a[data-color-map-component-id='#{color_map_file.perma_id}:aaa']")
         end
 
         it 'does not set data-color-map-component-id attribute if background type is hover_video' do
-          entry = create(:entry)
+          @entry = PublishedEntry.new(create(:entry, :published))
           configuration = {
             'linkmap_areas' => [{'color_map_component_id' => '1:aaa'}],
             'background_type' => 'hover_video'
           }
 
-          html = helper.linkmap_areas_div(entry, configuration)
+          html = helper.linkmap_areas_div(@entry, configuration)
 
           expect(html).not_to have_selector('a[data-color-map-component-id]')
         end
```

Comment:
We could consider adding a test helper to `pageflow-support` that can be called at the top of a spec to define a before hook that stubs `find_file_in_entry` even if `entry_has_file` is not called. That way there would not be a need to always set `@entry` even if the test case does not talk about files just because the helper under test calls `find_file_in_entry` somewhere deep inside.